.github/workflows/check-tools.py

```
@@ -0,0 +1,112 @@
+    try:
+        resp = requests.head(url, allow_redirects=True, timeout=10, headers=headers)
+        if resp.status_code >= 400:
+            return False, f"URL '{url}' returned HTTP status code {resp.status_code}"
+        return True, ""
+    except Exception as e:
+        return False, f"URL '{url}' could not be reached: {e}"
+
+def check_release(repo, version):
+    tag = f"v{version}"
+    url = f"{GITHUB_API}/repos/{repo}/releases/tags/{tag}"
+    resp = requests.get(url)
```

Ruff (0.15.0)

[error] Probable use of `requests` call without timeout

(S113)